pkg/vsphere/vm/vm.go

```diff
@@ -272,17 +272,26 @@ func (vm *VirtualMachine) DeleteExceptDisks(ctx context.Context) (*types.TaskInf
 
 	op := trace.FromContext(ctx, "DeleteExceptDisks")
 
+	op.Debugf("Getting list of the devices for VM %s", vm)
 	devices, err := vm.Device(op)
 	if err != nil {
 		return nil, err
 	}
 
 	disks := devices.SelectByType(&types.VirtualDisk{})
-	err = vm.RemoveDevice(op, true, disks...)
+	op.Debug("Removing devices for VM")
+
+	err = retry.Do(func() error {
+		return vm.RemoveDevice(op, true, disks...)
+	}, func(err error) bool {
+		return tasks.IsRetryError(op, err)
```

Comment:
Are we confident that we actually want to retry in all of these cases? Do we want to put any sort of bounding on this retry porcess? `tasks.IsRetryError` is really aggressive, in ways I'm not sure it should be (e.g., it retries things like `InvalidArgument` errors).

Reply:
> Do we want to put any sort of bounding on this retry process?

Nevermind; I see that `retry.Do` has a max.

Reply:
I've been bitten by retries of InvalidArgument before - I would like to know why that's in the retry set as it's not something that's likely to be transient. 
We should also look at allowing `tasks.IsRetryError` to be passed directly as a filter funciton to `retry.Do` seeing as they're likely to be used together.

Reply:
There's some information in #4597, but not a lot of specifics.

It's not clear how one would thoroughly test following the removal of `InvalidArgument` from `IsRetryError`, since it's not clear which places may be depending on that to avoid race-y behavior (e.g., it's not clear where we may attempt something knowing that it may fail until the system becomes consistent and the argument is no longer invalid). I suspect we don't have any sort of testing for this, since it doesn't seem like something that could be easily tested without a more complete/robust version of vcsim.